restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/OrganizationsEndpoint.java

```diff
@@ -0,0 +1,253 @@
+            } catch (Exception e) {
+                return status(NOT_FOUND).build();
+            }
+        }
+
+        if (organization == null) {
+            return status(NOT_FOUND).build();
+        } else {
+            if (APPLICATION_XML_TYPE == responseType) {
+                final RestCommResponse response = new RestCommResponse(organization);
+                return ok(xstream.toXML(response), APPLICATION_XML).build();
+            } else if (APPLICATION_JSON_TYPE == responseType) {
+                return ok(gson.toJson(organization), APPLICATION_JSON).build();
+            } else {
+                return null;
+            }
+        }
+    }
+
+    /**
+     * @param info
+     * @param responseType
+     * @return
+     */
+    protected Response getOrganizations(UriInfo info, final MediaType responseType) {
+        checkAuthenticatedAccount();
+        allowOnlySuperAdmin();
+
+        List<Organization> organizations = null;
+
+        String status = info.getQueryParameters().getFirst("Status");
+
+        if(status != null && Organization.Status.getValueOf(status.toLowerCase()) != null){
+            organizations = organizationsDao.getOrganizationsByStatus(Organization.Status.getValueOf(status.toLowerCase()));
+        }else{
+            organizations = organizationsDao.getAllOrganizations();
+        }
+        if (organizations == null || organizations.isEmpty()) {
+            return status(NOT_FOUND).build();
+        }
+        if (APPLICATION_XML_TYPE == responseType) {
+            final RestCommResponse response = new RestCommResponse(new OrganizationList(organizations));
+            return ok(xstream.toXML(response), APPLICATION_XML).build();
+        } else if (APPLICATION_JSON_TYPE == responseType) {
+            return ok(gson.toJson(organizations), APPLICATION_JSON).build();
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * putOrganization create new organization
+     * @param domainName
+     * @param data
+     * @param applicationJsonType
+     * @return
+     */
+    protected Response putOrganization(String domainName, MultivaluedMap<String, String> data,
+            MediaType responseType) {
+        if(domainName == null){
+            return status(BAD_REQUEST).entity(MSG_EMPTY_DOMAIN_NAME ).build();
+        }else{
+            checkAuthenticatedAccount();
+            allowOnlySuperAdmin();
+
+            //Character verification
+            final Pattern pattern = Pattern.compile(SUB_DOMAIN_NAME_VALIDATION_PATTERN);
+            if(!pattern.matcher(domainName).matches()){
+                return status(BAD_REQUEST).entity(MSG_INVALID_DOMAIN_NAME_PATTERN).build();
+            }
+
+            //Check if domain_name does not already taken inside restcomm by an organization.
+            Organization organization = organizationsDao.getOrganizationByDomainName(domainName);
+            if(organization != null){
+                return status(CONFLICT)
```

Comment:
i guess this checking its just a preValidation, anc concurrent access is later checked in backend,right? otherwise i dont see how this will enforce a unique cosntraint if concurrent accessed...

Reply:
this line is not here to prevent concurrency but instead its here to intend to see if domain name is not already taken by another organization in restcomm DB.
p.s. domain name column is unique in organization table so DB concurrency is handled at DB level